crates/solvers/src/infra/dex/paraswap/dto.rs

```diff
@@ -0,0 +1,223 @@
+    /// Source token address.
+    pub src_token: H160,
+
+    /// Destination token address.
+    pub dest_token: H160,
+
+    // Source amount.
+    #[serde_as(as = "serialize::U256")]
+    pub src_amount: U256,
+
+    // Destination amount.
+    #[serde_as(as = "serialize::U256")]
+    pub dest_amount: U256,
+
+    /// The decimals of the source token.
+    pub src_decimals: u8,
+
+    /// The decimals of the destination token.
+    pub dest_decimals: u8,
+
+    /// Price route from `/prices` endpoint response (without any change).
+    pub price_route: serde_json::Value,
+
+    /// The address of the signer.
+    pub user_address: H160,
+
+    /// The partner name.
+    pub partner: Option<String>,
+}
+
+impl TransactionBody {
+    pub fn new(
+        price: &Price,
+        config: &super::Config,
+        order: &dex::Order,
+        tokens: &auction::Tokens,
+        slippage: &dex::Slippage,
+    ) -> Result<Self, super::Error> {
+        let (src_amount, dest_amount) = match order.side {
+            order::Side::Sell => (price.src_amount()?, slippage.sub(price.dest_amount()?)),
+            order::Side::Buy => (slippage.add(price.src_amount()?), price.dest_amount()?),
+        };
+        Ok(Self {
+            src_token: order.sell.0,
+            dest_token: order.buy.0,
+            src_decimals: tokens
+                .decimals(&order.sell)
+                .ok_or(super::Error::MissingDecimals)?,
+            dest_decimals: tokens
+                .decimals(&order.buy)
+                .ok_or(super::Error::MissingDecimals)?,
+            src_amount,
+            dest_amount,
+            price_route: price.price_route.clone(),
+            user_address: config.address,
+            partner: config.partner.clone(),
+        })
+    }
+}
+
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum Side {
+    Sell,
+    Buy,
+}
+
+/// A ParaSwap API price response.
+#[serde_as]
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Price {
+    /// The price route. This should be passed on to the `/transactions`
+    /// endpoint.
+    pub price_route: serde_json::Value,
+}
+
+impl Price {
+    pub fn src_amount(&self) -> Result<U256, serde_json::Error> {
+        serde_json::from_value::<PriceRoute>(self.price_route.clone()).map(|r| r.src_amount)
+    }
+
+    pub fn dest_amount(&self) -> Result<U256, serde_json::Error> {
+        serde_json::from_value::<PriceRoute>(self.price_route.clone()).map(|r| r.dest_amount)
+    }
+
+    pub fn gas_cost(&self) -> Result<U256, serde_json::Error> {
+        serde_json::from_value::<PriceRoute>(self.price_route.clone()).map(|r| r.gas_cost)
+    }
+}
+
+#[serde_as]
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct PriceRoute {
+    #[serde_as(as = "serialize::U256")]
+    src_amount: U256,
+    #[serde_as(as = "serialize::U256")]
+    dest_amount: U256,
+    #[serde_as(as = "serialize::U256")]
+    gas_cost: U256,
```

Comment:
The `tokenProxyAddress` is also needed from the `priceRoute` response to get the approval target.